lib/ui/painting.dart

```diff
@@ -4960,6 +4960,22 @@ class ImmutableBuffer extends NativeFieldWrapperClass2 {
   }
   void _init(Uint8List list, _Callback<void> callback) native 'ImmutableBuffer_init';
 
+  /// Create a buffer from the asset with key [assetKey].
+  ///
+  /// Returns `null` if the asset does not exist.
+  static ImmutableBuffer? fromAsset(String assetKey) {
+    final ImmutableBuffer buffer = ImmutableBuffer._(0);
```

Comment:
I think we'd want to make this a factory so we could have `_fromAsset` return the length of the asset in bytes, and set the lenght properly.

Reply:
Alternatively, we could make the `length` a getter on a private field that is mutable.